### PR TITLE
[Contributing] [Standards] Do not use spaces inside/around offset accessors

### PR DESCRIPTION
**Pull request by phansys**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | N/A |

Ref: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2089

### CODE REVIEW

File: contributing/code/standards.rst [hunk 1]
```diff
@@ -176,7 +176,9 @@ Structure
   Read more at :ref:`contributing-code-conventions-deprecations`;
 
 * Do not use ``else``, ``elseif``, ``break`` after ``if`` and ``case`` conditions
-  which return or throw something.
+  which return or throw something;
+
+* Do not use spaces inside/around offset accessors;
```
Comment (keradus) [inline contributing/code/standards.rst]: should be full stop

**Comment by keradus** [comment]
codebase is following the rule (I checked 2.7 branch)
:+1: after fixing punctuation

**Comment by stof** [comment]
:+1: this is indeed a rule we apply

**Comment by phansys** [comment]
I'm thinking how to reword this rule in order to express what we really want, by instance:
"Do not use spaces around offset accessors (``[`` and ``]``), at least that their surrounding expressions are part of another separated expresion."
Otherwise, it can be currently interpreted as `$foo = []` is not good, same for `$foo[] = 'value'` or `$bar = ['foo' => ['baz]]`.

**Comment by keradus** [comment]
> Otherwise, it can be currently interpreted as `$foo = []`

here braces are not offset accessors, they are short array syntax

> $foo[] = 'value'

here, indeed, should be no spaces except space after `]`.


> `$bar = ['foo' => ['baz]]`

again short array syntax

----------------

so in general, sth like:
`Do not use spaces around `[` offset accessor and before `]` offset accessor.`

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

**Comment by wouterj** [comment]
Thanks Javier.

**Pull request closed without merging**